cmd/erasure-object.go

```diff
@@ -1706,8 +1708,21 @@ func (er erasureObjects) DeleteObjects(ctx context.Context, bucket string, objec
 	}
 
 	dedupVersions := make([]FileInfoVersions, 0, len(versionsMap))
-	for _, version := range versionsMap {
-		dedupVersions = append(dedupVersions, version)
+	for _, fivs := range versionsMap {
+		// Removal of existing versions and adding a delete marker in the same
+		// request is supported. At the same time, we cannot allow adding
+		// two delete markers on top of any object. To avoid this situation,
+		// we will sort deletions to execute existing deletion first,
+		// then add only one delete marker if requested
+		sort.SliceStable(fivs.Versions, func(i, j int) bool {
```

Comment:
Stable sort is only needed when you need to use for two different slices. You should just use sort.Slice() 

The API name is confusing sort.SliceStable() doesn't mean that sort.Slice is unstable.

Reply:
@harshavardhana It only sorts on a bool, so there will be a lot of ties. Therefore I would rather that it retains the order.

It is unstable as output in the two parts will be random. I would rather it just retain the order of these elements than jumble them up a random order.

Reply:
ah then it makes sense.